src/code/z_actor.c

```diff
@@ -97,29 +132,114 @@ GLOBAL_ASM("asm/non_matchings/z_actor//func_800B6608.asm")
 
 GLOBAL_ASM("asm/non_matchings/z_actor//func_800B6680.asm")
 
-GLOBAL_ASM("asm/non_matchings/z_actor//Actor_MarkForDeath.asm")
+void Actor_MarkForDeath(Actor* actor) {
+    actor->draw = NULL;
+    actor->main = NULL;
+    actor->flags &= -2;
+}
+
+void Actor_InitCurrPosition(Actor* actor) {
+    actor->currPosRot = actor->initPosRot;
+}
+
+void Actor_SetHeight(Actor* actor, f32 height) {
+    actor->topPosRot.pos.x = actor->currPosRot.pos.x;
+    actor->topPosRot.pos.y = actor->currPosRot.pos.y + height;
+    actor->topPosRot.pos.z = actor->currPosRot.pos.z;
+    actor->topPosRot.rot.x = actor->currPosRot.rot.x;
+    actor->topPosRot.rot.y = actor->currPosRot.rot.y;
+    actor->topPosRot.rot.z = actor->currPosRot.rot.z;
+}
+
+void Actor_SetRotationFromDrawRotation(Actor* actor) {
+    actor->currPosRot.rot = actor->drawParams.rot;
+}
+
+void Actor_InitDrawRotation(Actor* actor) {
+    actor->drawParams.rot = actor->currPosRot.rot;
+}
 
-GLOBAL_ASM("asm/non_matchings/z_actor//Actor_InitCurrPosition.asm")
+void Actor_SetScale(Actor* actor, f32 scale) {
+    actor->scale.z = scale;
+    actor->scale.y = scale;
+    actor->scale.x = scale;
+}
 
-GLOBAL_ASM("asm/non_matchings/z_actor//Actor_SetHeight.asm")
+void Actor_SetObjectSegment(GlobalContext* ctxt, Actor* actor) {
+    gRspSegmentPhysAddrs[6] = (s32) ctxt->sceneContext.objects[actor->objectIndex].vramAddr + 0x80000000;
+}
 
-GLOBAL_ASM("asm/non_matchings/z_actor//Actor_SetRotationFromDrawRotation.asm")
+#ifdef NON_MATCHING
+void Actor_InitToDefaultValues(Actor* actor, GlobalContext* ctxt) {
+    Actor_InitCurrPosition(actor);
+    Actor_InitDrawRotation(actor);
+    Actor_SetHeight(actor, 0);
+    Lib_CopyVec3f(&actor->lastPos, &actor->currPosRot.pos);
+    Actor_SetScale(actor, 0.01);
 
-GLOBAL_ASM("asm/non_matchings/z_actor//Actor_InitDrawRotation.asm")
+    actor->pad1F[0] = 3;
```

Comment:
If it's being used in a function, edit the struct to remove this padding (or reduce/split it up).

Reply:
Fixed